File: Multithreading/Laba3/Methods.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LabPP3
{
    class Methods
    {
        static object block = new object();
        private int sizeOfArrays;

        public Methods(int sizeOfArrays)
        {
            this.sizeOfArrays = sizeOfArrays;
        }

        public int[] FillingOfVector()
        {
            int[] vector = new int[sizeOfArrays];

            for (int i = 0; i < sizeOfArrays; i++)
            {
                vector[i] = 1;
            }

            return vector;
        }

        public int[,] FillingOfMatrix()
        {
            int[,] matrix = new int[sizeOfArrays, sizeOfArrays];

            for (int i = 0; i < sizeOfArrays; i++)
            {
                for (int j = 0; j < sizeOfArrays; j++)
                {
                    matrix[i, j] = 1;
                }
            }

            return matrix;
        }

        public void PutVector(int[] vector, String vectorName)
        {
            lock (block)
            {
                Console.WriteLine("Here is vector " + vectorName + " : ");
                for (int i = 0; i < sizeOfArrays; i++)
                {
                    Console.Write(vector[i] + " ");
                }

                Console.WriteLine();
            }
        }

        public void PutMatrix(int[,] matrix, String matrixName)
        {
            lock (block)
            {
                Console.WriteLine("Here is matrix " + matrixName + " : ");
                for (int i = 0; i < sizeOfArrays; i++)
                {
                    for (int j = 0; j < sizeOfArrays; j++)
                    {
                        Console.Write(matrix[i, j] + " ");
                    }
                    Console.WriteLine();
                }
            }
        }

        public int[] SumOfVector(int[] vector1, int[] vector2)
        {
            int[] resVector = new int[sizeOfArrays];

            for (int i = 0; i < sizeOfArrays; i++)
            {
                resVector[i] = vector1[i] + vector2[i];
            }

            return resVector;
        }
        
    }

        public int[] MultVectors(int[] vector1, int[] vector2){

            int value = 0;
            int[] resultVector = new int [vector1.Length];
            for(int i=0;i<vector1.Length;i++){
                for(int j=0;j<vector2.Length;i++){
                    value += vector1[i]*vector2[j];
                    resultVector[i] = value;
                }
                value = 0;
            }
            return resultVector;
        }

        public int[,] MultMatrixConst(int num, int[,] MyMatrix){


             for(int i=0;i<MyMatrix.Length;i++){
                for(int j=0;j<MyMatrix.Length;i++){
                    MyMatrix[i][j] = MyMatrix[i][j] * num;
                }
             }
             return MyMatrix;
        }

        public int MaxVector(int[] myVector){
            if(myVector.Length != 0){

                int maxElem = myVector[0];

                for(int i=1;i<myVector.Length;i++){

                    if(myVector[i] > maxElem){
                        maxElem = myVector[i];
                    }
                }
                return maxElem;
            }
            else{ return 0;}
        }


        public int[,] MultMatrix(int[,] matrix1, int[,] matrix2)
        {
            int[,] resultMatrix = new int[sizeOfArrays, sizeOfArrays];

            for (int i = 0; i < sizeOfArrays; i++)
            {
                for (int j = 0; j < sizeOfArrays; j++)
                {
                    for (int k = 0; k < sizeOfArrays; k++)
                    {
                        resultMatrix[i, j] += matrix1[i, k] * matrix2[k, j];
                    }
                }
            }

            return resultMatrix;
        }

    public int[] multipleMatrixVector(int[,] matrixToMultiple, int[] vectorToMultiple) {
        int[] result = new int[sizeOfArrays];
        for (int i = 0; i < sizeOfArrays; i++) {
            int sum = 0;
            for (int j = 0; j < sizeOfArrays; j++) {
                sum += matrixToMultiple[i, j] * vectorToMultiple[j];
            }
            result[i] = sum;
        }
        return result;
    }
    
}

File: Multithreading/Laba6final/Laba6final/Methods.cpp
#include "pch.h"
#include "Methods.h"


Methods::Methods(int sizeOfArrays)
{
	this->sizeOfArrays = sizeOfArrays;
}


Methods::~Methods()
{
}

//+
int* Methods::fillVector()
{
	int* vector = new int[sizeOfArrays];

	for (int i = 0; i < sizeOfArrays; i++)
	{
		//vector[i] = rand() % 100 - 100;
		vector[i] = 1;
	}

	return vector;
}
//+
int** Methods::fillMatrix()
{
	int** matrix = new int*[sizeOfArrays];

	for (int i = 0; i < sizeOfArrays; i++)
	{
		matrix[i] = new int[sizeOfArrays];
	}

	for (int i = 0; i < sizeOfArrays; i++)
	{
		for (int j = 0; j < sizeOfArrays; j++)
		{
			//matrix[i][j] = rand() % 100 - 100;
			matrix[i][j] = 1;
		}
	}

	return matrix;
}
//+
void Methods::putVector(int* vector)
{
	for (int i = 0; i < sizeOfArrays; i++)
	{
		std::cout << vector[i] << " ";
	}

	std::cout << '\n';
}
//+
void Methods::putMatrix(int** matrix)
{
	for (int i = 0; i < sizeOfArrays; i++)
	{
		for (int j = 0; j < sizeOfArrays; j++)
		{
			std::cout << matrix[i][j] << " ";
		}
		std::cout << '\n';
	}
}
//+
int* Methods::sumOfVectors(int* vector1, int* vector2)
{
	int* resVector = new int[sizeOfArrays];

	for (int i = 0; i < sizeOfArrays; i++)
	{
		resVector[i] = vector1[i] + vector2[i];
	}

	return resVector;
}

//+
int** Methods::multMatrix(int** matrix1, int** matrix2)
{
	int** resultMatrix = new int*[sizeOfArrays];

	for (int i = 0; i < sizeOfArrays; i++)
	{
		resultMatrix[i] = new int[sizeOfArrays];
	}

	for (int i = 0; i < sizeOfArrays; i++)
	{
		for (int j = 0; j < sizeOfArrays; j++)
		{
			resultMatrix[i][j] = 0;
		}

	}

	for (int i = 0; i < sizeOfArrays; i++) {
		for (int j = 0; j < sizeOfArrays; j++) {
			for (int k = 0; k < sizeOfArrays; k++) {
				resultMatrix[i][j] += matrix1[i][k] * matrix2[k][j];
			}
		}
	}

	return resultMatrix;
}
int** Methods::diffMatrix(int** matrix1, int** matrix2)
{
	int** resultMatrix = new int*[sizeOfArrays];

	for (int i = 0; i < sizeOfArrays; i++)
	{
		resultMatrix[i] = new int[sizeOfArrays];
	}

	for (int i = 0; i < sizeOfArrays; i++) {
		for (int j = 0; j < sizeOfArrays; j++) {
			resultMatrix[i][j] = matrix1[i][j] - matrix2[i][j];
		}
	}

	return resultMatrix;
}

int* Methods::MultMatrixVector(int** matrixToMultiple, int* vectorToMultiple) {
	int* result = new int[sizeOfArrays];
	for (int i = 0; i < sizeOfArrays; i++) {
		int sum = 0;
		for (int j = 0; j < sizeOfArrays; j++) {
			sum += matrixToMultiple[i][j] * vectorToMultiple[j];
		}
		result[i] = sum;
	}
	return result;
}
//+
int** Methods::transMatrix(int** matrix)
{
	int** transMatr = new int*[sizeOfArrays];

	for (int i = 0; i < sizeOfArrays; i++)
	{
		transMatr[i] = new int[sizeOfArrays];
	}

	for (int i = 0; i < sizeOfArrays; i++)
	{
		for (int j = 0; j < sizeOfArrays; j++)
		{
			transMatr[i][j] = 0;
		}
	}

	for (int i = 0; i < sizeOfArrays; i++) {
		for (int j = 0; j < sizeOfArrays; j++) {
			transMatr[i][j] = matrix[j][i];
		}
	}
	return transMatr;
}